src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,197 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = tail = new Node<>(null, value, null);
+        } else {
+            tail = tail.next = new Node<>(tail, value, null);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        addByIndexRoutine(value, index);
```

Comment:
let's just remove the private method and move it's logic to here instead, because the signature of the method is identical and it contains only a call to the private one, thus can be a public, it is also used only here